### ISSUE TITLE
False positive in unnecessarry_lambda_linter()

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/lintr#2310]
``` r
library(lintr)

lint(text = "lapply(l, function(e) 'a' %in% names(e))", linters = unnecessary_lambda_linter())
#> <text>:1:11: warning: [unnecessary_lambda_linter] Pass names directly as a symbol to lapply() instead of wrapping it in an unnecessary anonymous function. For example, prefer lapply(DF, sum) to lapply(DF, function(x) sum(x)).
#> lapply(l, function(e) 'a' %in% names(e))
#>           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2023-11-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
can you confirm this is in the current dev version? there's been some recent related fixes

**Comment by IndrajeetPatil** [comment]
I had actually checked with current dev before adding the label. This indeed seems like another false positive.

**Comment by Bisaloo** [comment]
Fixed by #2320
